code/game/objects/items/weapons/switchtool.dm

```diff
@@ -207,3 +216,178 @@
+		var/datum/tech/disk_tech = T.stored
+		if(istype(disk_tech, /datum/tech/powerstorage) && disk_tech.level >= 4)
+			stored_modules["/obj/item/weapon/soap:Soap"] = new /obj/item/weapon/circular_saw(src)
+			to_chat(user, "The Holo switchtool has power clean designs now!")
+			return
+	**/
+//Since you can't turn off the welder inside the tool, I'm using the unused welder that very slowly regens fuel, looks like 1u per 5 byond seconds, thanks byond.
+//It can be refulled manually, but since it starts active you will blow up welder tanks if deployed and then put to a tank.
+	if(istype(deployed, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/weldingtool = deployed
+		weldingtool.welding = 1
+		weldingtool.status = 1
+		weldingtool.max_fuel = 50
+		weldingtool.start_fueled = 1
+
+//for the inhand sprite changes
+//Big shitty wall of else if lazy code
+
+obj/item/weapon/switchtool/holo/update_icon()
+	if(istype(deployed, /obj/item/device/flashlight))
+		item_state = "flashlight"
+
+	else if(istype(deployed, /obj/item/weapon/scalpel))
+		item_state = "scalpel"
+
+	else if(istype(deployed, /obj/item/weapon/circular_saw))
+		item_state = "circularsaw"
+
+	else if(istype(deployed, /obj/item/weapon/surgicaldrill))
+		item_state = "drill"
+
+	else if(istype(deployed, /obj/item/weapon/cautery))
+		item_state = "cautery"
+
+	else if(istype(deployed, /obj/item/weapon/hemostat))
+		item_state = "hemostat"
+
+	else if(istype(deployed, /obj/item/weapon/retractor))
+		item_state = "retractor"
+
+	else if(istype(deployed, /obj/item/weapon/bonesetter))
+		item_state = "bonesetter"
+
+	else if(istype(deployed, /obj/item/weapon/screwdriver))
+		item_state = "screwdriver"
+
+	else if(istype(deployed, /obj/item/weapon/wrench))
+		item_state = "wrench"
+
+	else if(istype(deployed, /obj/item/weapon/wirecutters))
+		item_state = "wirecutters"
+
+	else if(istype(deployed, /obj/item/weapon/crowbar))
+		item_state = "crowbar"
+
+	else if(istype(deployed, /obj/item/device/multitool))
+		item_state = "multitool"
+
+	else if(istype(deployed, /obj/item/weapon/weldingtool))
+		item_state = "weldingtool"
+
+	else if(istype(deployed, /obj/item/weapon/shield/energy))
+		item_state = "shield"
+
+	else if(istype(deployed, /obj/item/weapon/melee/energy/sword/activated))
+		item_state = "sword"
+
+	else if(istype(deployed, /obj/item/weapon/melee/energy/hfmachete/activated))
+		item_state = "sharper sword"
+
+	//else if(istype(deployed, /obj/item/weapon/soap))
+		//item_state = "soap"
+
+	else
+		item_state = "Hswitchtool"
+
+	if(istype(loc, /mob))
+		var/mob/M = loc
+		M.update_inv_hands()
+
+/obj/item/weapon/switchtool/holo/IsShield()
+	if(istype(deployed, /obj/item/weapon/shield/energy))
+		return 1
+	else
+		return 0
+
+//All modules make small amounts of light, flashlight making more.
+/obj/item/weapon/switchtool/holo/deploy(var/module)
+	if(!..())
+		return FALSE
+	set_light(brightness_min)
+	if(istype(deployed, /obj/item/device/flashlight))
```

Comment:
Can it not just use the light from the flashlight itself? I'm not sure if lighting works that way.
But either way, it should use the flashlight's brightness.